sqlglot/expressions.py

```diff
@@ -274,12 +274,16 @@ def append(self, arg_key: str, value: t.Any) -> None:
 
     def set(self, arg_key: str, value: t.Any) -> None:
         """
-        Sets `arg_key` to `value`.
+        Sets arg_key to value.
 
         Args:
-            arg_key (str): name of the expression arg.
+            arg_key: name of the expression arg.
             value: value to set the arg to.
         """
+        if value is None:
+            self.args.pop(arg_key, None)
+            return
```

Comment:
I noticed the following unintuitive (imo) behaviour when setting an arg to `None`, which motivated this change:

```python
>>> import sqlglot
>>> expr = sqlglot.parse_one("select * from foo join bar")
>>> expr.set("joins", None)
>>> type(expr.args.get("joins", []))
<class 'NoneType'>
```

The issue here is that setting a dict entry to `None` does not have any additional meaning like "deleting the expression". It simply results in a dict entry being set to `None`, so in this case `"joins" -> None` and as shown above this doesn't play well with the semantics of the `get` method on dicts.

Do we ever want to maintain a `None` arg in an expression without deleting it? If an expression is optional, we'll use the `.get(...)` call anyway and it'll return `None`.

FYI: this led to some test failures because we use the `for join in expr.args.get("joins", [])` pattern in the codebase.

Reply:
Perhaps we could also add an optional `arg_key` parameter to the `pop` method as an alternative? It currently works only on expressions, so we can't do `expr.args["joins"].pop()`, but we could instead have `expr.pop("joins")`

Reply:
that's why you shouldn't do get("", []) and always do get("") or []